src/renderer/components/ft-toast/ft-toast.js

```diff
@@ -0,0 +1,54 @@
+import Vue from 'vue'
+import FtToastEvents from './ft-toast-events.js'
+
+export default Vue.extend({
+  name: 'FtToast',
+  data: function () {
+    return {
+      toasts: [
+      ],
+    }
+  },
+  mounted: function () {
+    FtToastEvents.$on('toast.open', this.open)
+  },
+  beforeDestroy: function () {
+    FtToastEvents.$off('toast.open', this.open)
+  },
+  methods: {
+    performAction: function (toast) {
+      toast.action()
+      this.close(toast)
+    },
+    close: function (toast) {
+      clearTimeout(toast.timeout)
+      // Remove toasts when most recent toast has finished to avoid re-render
+      if (this.toasts.filter(toast => toast.isOpen).length === 1) {
+        // Wait for fade-out to finish
+        setTimeout(this.clear, 300)
+      }
+      toast.isOpen = false
+    },
+    open: function (message, action, time) {
+      const toast = { message: message, action: action || (() => { }), isOpen: false, timeout: null }
+      toast.timeout = setTimeout(this.close, time || 3000, toast)
+      setImmediate(() => { toast.isOpen = true })
+      if (this.toasts.length > 4) {
```

Comment:
This if statement can be simplified.  We want it so that when a new toast is sent and the array is already at 5, the last toast is pushed out of the array and the new one will be inserted, while always taking the bottom slot.  Simply splicing out the first object in the array will accomplish this.

```
if (this.toasts.length > 4) {
    this.toasts.splice(0, 1)
}
```